### PR TITLE
Add queryChildren and queryChildrenAll

### PR DESCRIPTION
**Pull request by rtfeldman**
discussed w/ @eeue56

### CODE REVIEW

File: src/ElmHtml/Query.elm [hunk 1]
```diff
@@ -96,15 +98,39 @@ queryAll selectors =
...
 {-| Query a Html node using a selector
 -}
 queryInNode : Selector -> ElmHtml -> List ElmHtml
-queryInNode selector node =
+queryInNode =
+    queryInNodeHelp True
+
+
+queryInNodeHelp : Bool -> Selector -> ElmHtml -> List ElmHtml
+queryInNodeHelp recurse selector node =
```
Comment (eeue56) [inline src/ElmHtml/Query.elm]: :art: I think recurse isn't a very clear name - it doesn't explain what it does. `shouldLookInChildren` might be clearer

File: src/ElmHtml/Query.elm [hunk 2]
```diff
@@ -96,15 +98,39 @@ queryAll selectors =
...
+{-| Query to ensure a html node has all selectors given, without considering
+any descendants lower than its immediate children.
+-}
+queryChildrenAll : List Selector -> ElmHtml -> List ElmHtml
+queryChildrenAll selectors =
+    queryInNodeHelp False (Multiple selectors)
+
+
 {-| Query a Html node using a selector
 -}
```
Comment (eeue56) [inline src/ElmHtml/Query.elm:121]: this comment should be updated to say that it goes all the way down

Comment (rtfeldman) [reply src/ElmHtml/Query.elm:121]: done!

File: src/ElmHtml/Query.elm (hunk 1 above)
Comment (rtfeldman) [reply src/ElmHtml/Query.elm]: revised to make it both a bit more flexible and also hopefully clearer!

File: src/ElmHtml/Query.elm [hunk 3]
```diff
@@ -104,42 +104,53 @@ any descendants lower than its immediate children.
...
     case node of
         NodeEntry record ->
             let
-                mapChildren children =
-                    if recurse then
-                        List.concatMap (queryInNode selector) children
-                    else
-                        []
+                childEntries =
+                    case maxDescendantDepth of
```
Comment (eeue56) [inline src/ElmHtml/Query.elm]: this nesting is getting a bit deep, maybe switch it out for `Maybe.map`

File: src/ElmHtml/Query.elm [hunk 4]
```diff
@@ -92,15 +92,16 @@ query selector =
...
 -}
 queryAll : List Selector -> ElmHtml -> List ElmHtml
 queryAll selectors =
     query (Multiple selectors)
 
 
-{-| Query a Html element using a selector, without considering
-any descendants lower than its immediate children.
+{-| Query a Html node using a selector, considering both the node itself
+as well as all of its descendants.
```
Comment (eeue56) [inline src/ElmHtml/Query.elm:104]: 💯

File: src/ElmHtml/Query.elm (hunk 3 above)
Comment (rtfeldman) [reply src/ElmHtml/Query.elm]: Good point! I think the branches are a bit too different for `Maybe.map` to be nice, so I split it into a helper function in c46d68f25988541dede484317bd15ec71cc9c53b

**Comment by eeue56** [comment]
@rtfeldman can you give me a timeline roughly on when you need this merged by?

**Comment by rtfeldman** [comment]
I need it for 1.0.0 of elm-html-test, but that also needs the pretty
printing PR to be merged and the whitelist PR to be merged, so it's not
super urgent.

Sometime in the next week or two would be sweet I guess?

**Pull request merged**